app/resources/applications/ApplicationsController.jl

```diff
@@ -409,7 +413,22 @@ function purge(app)
 end
 
 function uuid()
-  (:uuid => Genie.Secrets.secret_token()[end-11:end]) |> json
+  uuidstore_filepath = joinpath(@get_scratch!(GB_SCRATCH_SPACE_NAME), UUIDSTORE_FILENAME)
+
+  if !isfile(uuidstore_filepath)
```

Comment:
This needs to be in try/catch - if exception, just return the uuid without persisting